### PR TITLE
Fix circular structure _dequeBatch issue halting logging

### PR DESCRIPTION
**Pull request by Cheesier**
Including test that reproduces error.

Issue https://github.com/datalust/seq-logging/issues/24 describes issue in a little more detail.

### CODE REVIEW

File: test/seq_logger_tests.js [hunk 1]
```diff
@@ -111,8 +111,21 @@ describe('SeqLogger', () => {
...
+       it('does handle event properties with circular structures', () => {
+        let logger = new SeqLogger({serverUrl: 'https://my-seq/prd', apiKey: '12345'});
+        const event = makeCircularTestEvent();
+        logger.emit(event);
+        try {
+            logger.flushToBeacon();
+        }
+        finally {
+            logger._wipeQueue();
+        }
```
Comment (larenelg) [inline test/seq_logger_tests.js:123]: Just wondering what your thoughts are behind `_wipeQueue()`. The queue should be empty after line 119 if `flushToBeacon` succeeds. If `flushToBeacon` fails should we catch that error?

**Review by larenelg — APPROVED** [review]
Just waiting on your response to if `_wipeQueue` is still required :)

The rest LGTM! Thanks! 👍

File: test/seq_logger_tests.js (hunk 1 above)
Comment (Cheesier) [reply test/seq_logger_tests.js]: wipeQueue Is not needed no, fixing in next commit :)

**Comment by Cheesier** [comment]
Made some changes to how the circularRemover worked, it had an issue where it could think things are circular when they are in fact not.

Also log an error when the circularRemover is used, to make it easier for developers to see that it is happening.

**Comment by larenelg** [comment]
Thanks! LGTM, will merge :)

**Pull request merged**